torch/_functorch/aot_autograd.py

```diff
@@ -2590,24 +2590,11 @@ def backward(ctx, *flat_args):
             def call_compiled_backward():
                 if CompiledFunction.compiled_bw is None:
                     assert all(a is not None for a in all_args)
-                    if aot_config.dynamic_shapes:
-                        all_args_list = list(all_args)
-                        CompiledFunction.compiled_bw = create_aot_dispatcher_function(
-                            bw_module, all_args_list, AOTConfig(
-                                aot_config.bw_compiler, None, None,
-                                aot_config.decompositions, 0, aot_config.aot_id,
-                                aot_config.keep_inference_input_mutations,
-                                aot_config.dynamic_shapes,
-                                inference_compiler=None,
-                                aot_autograd_arg_pos_to_source=None,
-                            )
+                    context = disable_autocast_manager if disable_amp else nullcontext
+                    with context(), track_graph_compiling(aot_config, "backward"):
+                        CompiledFunction.compiled_bw = aot_config.bw_compiler(
+                            bw_module, all_args
```

Comment:
This should also fix https://github.com/pytorch/pytorch/issues/98416 ?

Reply:
Yep!